### ISSUE TITLE
Database design

### ISSUE THREAD

**Issue by epageperron** [cdli-gh/Framework#15]
## Summary
In order to make the CDLI Platform more efficient, we need to completely redesign the database, optimizing it based on the data.  

Design a fully relational model to store catalogue, textual data and new data from MTAAC. 

## Tasks
- [x] Explore CakePHP3 by doing the tutorials (w1)
- [x] Design a draft of the catalogue portion of the database (w2)
( Checkout Cakephp3 Database conventions here <https://book.cakephp.org/3.0/en/intro/conventions.html>
    - [x]  For this, first review <https://docs.google.com/spreadsheets/d/1FyYuZJlyFbd-NdWKyGXasY1iS4h4JKaTfPactMVn6Ok/edit#gid=0> with @epageperron
    - [x] Second, use PHPMyAdmin recommendations tool to get a better sense of the repeating text data
    - [x] Create the main table for artifacts
    - [x] Prepare relational tables
    - [x] Set up indexes and keys
- [x] Review and correction of the draft after consultation with @epageperron, @greek-geek, @starlordphr @zarpanitu and @englundrk @amanbis  (w2/3)
- [x] Design bibliographic section + relationships (w3)
    - [ ] Include a World Cart field for OCLC # (eg http://www.worldcat.org/oclc/89291834  ---> 89291834 )
    - [ ] See if http://tronprog.blogspot.ca/2010/05/phpbibliography-10-using-cakephp.html can be updated, tunkated and adapted
    - [ ] Or this https://libraries.io/packagist/audiolabs%2Fbibtexparser or https://libraries.io/packagist/renanbr%2Fbibtex-parser ...
- [x] Design corpus tool section if required (w3) (#)
- [x] Script a data migration tool from old to new schema (w4)
- [x] Bake and review the cakephp3 model layer to connect with the database (w4)

## Other links or relevant information
This task is dependant on 2 things: 
1) the  FM Challenge (#16)
2) The requirements of a corpus analysis tool

After this issue is done, we will have a full functional first draft of the DB but we might still have to make later adjustments to the schema and thus also to the conversion script and the model layer of Cake.

Other notes : 
Don't forget to take into account revisions and retired P nos.
Store CDLI-CoNLL version of ATF
Bool version consistent or not
Bool uncertain for period and provenience

**Everything that will have a uri should have a specific field**
Rdb2rdf for the LOD: http://www.rdb2rdf.org/

#### 📈 Progress (0-1): 0.3

See Gantt: http://cdli-dev.org/gantt/Framework/

**Comment by epageperron** [comment]
Research to do in parallel, integrate requirements in the proces:
Store annotated version of text Connl
Bool version consistent or not
Everything that will have a uri should have a specific field
Rdb2rdf for the LOD: http://www.rdb2rdf.org/
Reuse an existing corpus tool: (TO TEST (on cdli), do they scale?)
https://lindat.mff.cuni.cz/services/pmltq/#!/home ?
http://corpus-tools.org/annis/ ?
the Ziggurat? <3 
XML DB with own SVG frontend ?
SPARQL endpoint with visualization ?
Other possibilities ?

**Comment by epageperron** [comment]
Bibliographic data management has been added to #79, a new issue.

Most of the work in this issue has been completed, @kbabu105 will be finishing up the conversion and scripts  during the summer.
